### PR TITLE
[samples] Replace dictionary calls by RestDictionaryService

### PR DESCRIPTION
**Pull request by illiakovalenko**
## Description / Motivation
Replace dictionary calls by RestDictionaryService

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update (non-breaking change; modified files are limited to the `/docs` directory)

### CODE REVIEW

File: samples/node-headless-ssr-proxy/config.js [hunk 1]
```diff
@@ -141,25 +149,14 @@ const config = {
...
-
-        dictionaryCache.set(cacheKey, viewBag);
-        return viewBag;
-      });
+    return dictionaryService.fetchDictionaryData(language).then(phrases => {
+      const viewBag = {
+        dictionary: phrases,
+      };
+
+      dictionaryCache.set(cacheKey, viewBag);
```
Comment (ambrauer) [inline samples/node-headless-ssr-proxy/config.js:157]: Our `RestDictionaryService` includes caching, so we should be able to get rid of this caching logic and the dependency on `node-cache`. Maybe we should explicitly set cache options (using same TTL value) in the `new RestDictionaryService` above to make it clear?

File: samples/node-headless-ssr-proxy/config.js [hunk 2]
```diff
@@ -141,25 +149,14 @@ const config = {
 
     if (cached) return Promise.resolve(cached);
 
-    return fetch(
```
Comment (ambrauer) [inline samples/node-headless-ssr-proxy/config.js:144]: Looks like we can get rid of `node-fetch` dependency.

**Review by ambrauer — CHANGES_REQUESTED** [review]
See comments. Also, looks like we need to remove doc references to `i18next-fetch-backend` here: https://jss.sitecore.com/docs/fundamentals/services/dictionary-service

File: samples/node-headless-ssr-proxy/config.js (hunk 1 above)
Comment (illiakovalenko) [reply samples/node-headless-ssr-proxy/config.js:157]: Good catch! I forgot that we have default caching...

File: samples/node-headless-ssr-proxy/config.js (hunk 2 above)
Comment (illiakovalenko) [reply samples/node-headless-ssr-proxy/config.js:144]: It's true

**Review by ambrauer — APPROVED** [review]
Looks good!

**Pull request merged**